icaruscode/Filters/FilterCRTPMTMatching_module.cc

```diff
@@ -13,556 +13,607 @@
+    fMatchTree->Branch("gate_start_timestamp", &m_gate_start_timestamp,
+                       "gate_start_timestamp/l");
+    fMatchTree->Branch("trigger_gate_diff", &m_trigger_gate_diff,
+                       "trigger_gate_diff/l");
+  }
 }
 
-bool icarus::crt::FilterCRTPMTMatching::filter(art::Event &e)
-{
-    mf::LogDebug("FilterCRTPMTMatching: ") << "beginning analyis";
-
-    // Start by fetching some basic event information for our n-tuple.
-    fEvent = e.id().event();
-    fRun = e.run();
-    fSubRun = e.subRun();
-
-    ClearVecs();
-
-    // add trigger info
-    auto const& trigger_handle = e.getProduct<sbn::ExtraTriggerInfo>(fTriggerLabel);
-    sbn::triggerSource bit = trigger_handle.sourceType;
-    m_gate_type = value(bit);
-    m_gate_name = bitName(bit);
-    m_trigger_timestamp = trigger_handle.triggerTimestamp;
-    m_gate_start_timestamp = trigger_handle.beamGateTimestamp;
-    m_trigger_gate_diff = trigger_handle.triggerTimestamp - trigger_handle.beamGateTimestamp;
-
-    //Read Beam Gate Size
-    art::Run const& run = e.getRun();
-    auto const& trigger_configuration = run.getProduct<icarus::TriggerConfiguration>(fTriggerConfiguration);
-    m_gate_width=trigger_configuration.getGateWidth(m_gate_type);
-
-    // CRTHits
-
-    auto const& crtHitList = e.getProduct<std::vector<CRTHit>>(fCrtHitModuleLabel);
-
-    bool Cosmic = false;
-    int  Type   = 9;
-
-    if (fFilterLevel != "loose" || fFilterLevel != "tight" || fFilterLevel != "medium") fFilterLevel="loose";
-
-    for (auto const& flashLabel : fOpFlashModuleLabelVec)
-    {
-        art::Handle<std::vector<recob::OpFlash>> flashHandle;
+void icarus::crt::FilterCRTPMTMatching::getTriggerConf(art::Run& r) {
+  fTriggerConfiguration =
+      r.getProduct<icarus::TriggerConfiguration>(fTriggerConfigurationLabel);
+}
 
-        if (!e.getByLabel(flashLabel, flashHandle))
-        {
-            mf::LogError("FilterCRTPMTMatching") << "Did not find opflash object with label: " << flashLabel;
-            continue;
+bool icarus::crt::FilterCRTPMTMatching::filter(art::Event& e) {
+  mf::LogDebug("FilterCRTPMTMatching: ") << "beginning analyis";
+
+  // Start by fetching some basic event information for our n-tuple.
+  fEvent = e.id().event();
+  fRun = e.run();
+  fSubRun = e.subRun();
+
+  ClearVecs();
+  // add trigger info
+  if (!fTriggerLabel.empty()) {
+    art::Handle<sbn::ExtraTriggerInfo> trigger_handle;
+    e.getByLabel(fTriggerLabel, trigger_handle);
+    if (trigger_handle.isValid()) {
+      sbn::triggerSource bit = trigger_handle->sourceType;
+      m_gate_type = (unsigned int)bit;
+      m_gate_name = bitName(bit);
+      m_trigger_timestamp = trigger_handle->triggerTimestamp;
+      m_gate_start_timestamp = trigger_handle->beamGateTimestamp;
+      m_trigger_gate_diff =
+          trigger_handle->triggerTimestamp - trigger_handle->beamGateTimestamp;
+      m_gate_width = fTriggerConfiguration.getGateWidth(m_gate_type);
```

Comment:
Unless I have missed it, `fTriggerConfiguration` is not initialised anywhere (`getTriggerConf()` is not called).
The best place to call it is in the `beginRun()` method:
```cpp
void icarus::crt::FilterCRTPMTMatching::beginRun(art::Run& run) {
  getTriggerConf(run);
}
```
which need to be also declared in the class:
```cpp
    void beginRun(art::Run& run) override;
```